### PR TITLE
bump installer

### PR DESCRIPTION
**Pull request by jim-minter**
revendor installer to latest 4.3 + carry patches
upstream installer branch is https://github.com/jim-minter/installer/tree/release-4.3-azure-inprogress until this is e2e'd

e2e is waiting on new rhcos image publish (currently in flight)

@mjudeikis fyi

### CODE REVIEW

File: pkg/install/1-installresources.go [hunk 1]
```diff
@@ -320,17 +320,17 @@ func (i *Installer) installResources(ctx context.Context) error {
...
-											ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/backendAddressPools', '" + infraID + "-internal-lb', '" + infraID + "-internal-controlplane')]"),
+											ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/backendAddressPools', '" + infraID + "-internal-lb', '" + infraID + "-internal-controlplane-v4')]"),
 										},
 									},
 									Subnet: &mgmtnetwork.Subnet{
 										ID: to.StringPtr(i.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID),
 									},
 								},
-								Name: to.StringPtr("bootstrap-nic-ip"),
+								Name: to.StringPtr("bootstrap-nic-ip-v4"),
```
Comment (mjudeikis) [inline pkg/install/1-installresources.go:333]: This means all existing cluster will never get ipv6 :/ or will get it in a strange hybrid snowflaky form...

File: vendor/github.com/openshift/installer/pkg/asset/manifests/azure/cloudproviderconfig.go [hunk 2]
```diff
@@ -55,8 +53,6 @@ func (params CloudProviderConfig) JSON() (string, error) {
 
 	if params.ARO {
 		config.authConfig.UseManagedIdentityExtension = false
```
Comment (mjudeikis) [inline vendor/github.com/openshift/installer/pkg/asset/manifests/azure/cloudproviderconfig.go:55]: Bumpt for anybody review'ing. Carry on change.

File: vendor/github.com/openshift/installer/pkg/asset/manifests/cloudproviderconfig.go [hunk 3]
```diff
@@ -29,7 +32,10 @@ import (
 )
 
 var (
-	cloudProviderConfigFileName = filepath.Join(manifestDir, "cloud-provider-config.yaml")
+	cloudProviderConfigFileName         = filepath.Join(manifestDir, "cloud-provider-config.yaml")
```
Comment (mjudeikis) [inline vendor/github.com/openshift/installer/pkg/asset/manifests/cloudproviderconfig.go:35]: Bump for anybody review'ing. Carry on change. Delta from upstream

File: vendor/github.com/openshift/installer/pkg/asset/manifests/infrastructure.go [hunk 4]
```diff
@@ -149,7 +149,7 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 	if cloudproviderconfig.ConfigMap != nil {
 		// set the configmap reference.
 		config.Spec.CloudConfig = configv1.ConfigMapFileReference{Name: cloudproviderconfig.ConfigMap.Name, Key: cloudProviderConfigDataKey}
-		i.FileList = append(i.FileList, cloudproviderconfig.File)
+		i.FileList = append(i.FileList, cloudproviderconfig.Files()...)
```
Comment (mjudeikis) [inline vendor/github.com/openshift/installer/pkg/asset/manifests/infrastructure.go:152]: Bump for anybody review'ing. Carry on change.

File: pkg/install/1-installresources.go (hunk 1 above)
Comment (jim-minter) [reply pkg/install/1-installresources.go:333]: I worked through these changes - all they do is keep us close to the installer, I don't think they limit our options (further).  If we do IPv6 at some point it is probably already likely that we would not try to retrofit it to existing clusters.

**Pull request merged**